### PR TITLE
Add support for Micro metrics

### LINKED ISSUE #9: Supporting multiple metric of different type
The goal is to support multiple metrics with different goals (max/min)

### LINKED ISSUE #19: Report micro/macro-level metrics
let C_i be the number of correct examples in the i-th class
let T_i be the total number of examples in the i-th class

macro : (C_1 / T_1, C_2/ T_2, ...)
micro : (C_1 + C_2 + ...) / (T_1 + T_2 + ...)

### PR DESCRIPTION
**Pull request by ljj7975**
Honk2 now supports MACRO and MICRO metrics
Also, it also allows user to specify a metric that user wants to optimize for
Details can be found in #19 and #9 

close #9
close #19

### CODE REVIEW

File: metric/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from .acc import Acc
+
+from .per_class_acc import PerClassAcc
```
Comment (daemon) [inline metric/__init__.py:2]: I don't think double-spacing is necessary.

File: metric/acc.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import torch
+
+from .metric_utils import MacroMetric
+from utils import register_cls
+
+
+@register_cls('metric.Acc')
+class Acc(MacroMetric):
```
Comment (daemon) [inline metric/acc.py]: This is a micro-average, not a macro-average.

File: metric/metric_utils.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        pass
+
+
+class MacroMetric(Metric):
+    def __init__(self):
+        self.type = MetricType.MACRO
+
+    def get_type(self):
+        return self.type
+
```
Comment (daemon) [inline metric/metric_utils.py:29]: +1 \n here.

**Review by daemon — APPROVED** [review]
The definition of micro- vs macro- is flipped.
